rs/ic-observability/service-discovery/src/registry_sync.rs

```diff
@@ -182,8 +175,14 @@ async fn get_nns_public_key(
         )
         .await
         .map_err(|e| anyhow::format_err!("failed to get public key: {}", e))?;
-    Ok(ThresholdSigPublicKey::try_from(
-        PublicKey::decode(nns_pub_key_vec.as_slice()).expect("invalid public key"),
+    Ok(
+        ThresholdSigPublicKey::try_from(PublicKey::decode(nns_pub_key_vec.as_slice()).expect("invalid public key"))
+            .expect("failed to create thresholdsig public key"),
     )
-    .expect("failed to create thresholdsig public key"))
+}
+
+pub async fn ping_nns(nns_urls: Vec<Url>) -> bool {
```

Comment:
The name of this function is a bit misleading since `ping` normally means sending an ICMP packet to the destination. How about naming it `nns_reachable`?